tests/cocotb/basic/test/test_hwpe_stream_merge.py

```diff
@@ -0,0 +1,305 @@
+#---------------------------------
+# Copyright 2023 KULeuven
+# Solderpad Hardware License, Version 0.51, see LICENSE for details.
+# SPDX-License-Identifier: SHL-0.51
+# Author: Ryan Antonio (ryan.antonio@esat.kuleuven.be)
+#---------------------------------
+
+#-----------------------------------
+# Importing useful tools 
+#-----------------------------------
+import os
+import yaml
+import random
+import math
+import sys
+
+#-----------------------------------
+# Importing cocotb 
+#-----------------------------------
+import  cocotb
+from    cocotb.triggers       import RisingEdge, Timer
+from    cocotb.clock          import Clock
+from    cocotb_test.simulator import run
+
+#-----------------------------------
+# Importing pytest
+#-----------------------------------
+import  pytest
+
+#-----------------------------------
+# Extracting and setting important variables and paths
+#-----------------------------------
+hwpe_stream_path = os.getenv("HWPE_STREAM_HOME")
+basic_path       = hwpe_stream_path + "/tests/cocotb/basic"
+
+#-----------------------------------
+# Top-level definitions 
+#-----------------------------------
+# Specify top-level module
+toplevel     = 'wrapper_hwpe_stream_merge'
+# Specify python test name that contains the @cocotb.test. Usually the name of this test.
+module       = "test_hwpe_stream_merge"
+# Specify what simulator to use (e.g., verilator, modelsim, icarus)
+simulator    = "verilator"
+# Specify build directory
+sim_build    = basic_path + "/test/sim_build/{}/".format(toplevel)
+# Get YAML source files path
+src_yml_path = hwpe_stream_path + "/src_files.yml"
+
+#-----------------------------------
+# Global parameters for testing
+# TODO: These are modifiable so change whenever needed
+#-----------------------------------
+# Checker parameter
+CHECK_COUNT   = 5
+
+# DUT parameters
+NB_IN_STREAMS = 4
+DATA_WIDTH    = 16
+
+# For random seed logging
+RANDOM_SEED = random.randrange(sys.maxsize)
+random.seed(RANDOM_SEED)
+
+
+#-----------------------------------
+# Get YAML source
+#-----------------------------------
+with open(src_yml_path,"r") as src_files:
+    try:
+        yaml_dict = yaml.safe_load(src_files)
+    except yaml.YAMLError as exc:
+        print(exc)
+
+#-----------------------------------
+# Extract source from YAML
+# Also append the main working path
+#-----------------------------------
```

Comment:
I still don't understand why we need the yaml file for this?
This seems like an unnecessary extra level of indirection for this test setup?
What does the yaml file have of extra information?

Can't we just use some of these built-in methods instead (code example produced by google bard)?
```python
import os

def list_files():
  """Lists all files in the current directory."""
  files = os.listdir()
  for file in files:
    print(file)

if __name__ == "__main__":
  list_files()
```

Reply:
Eitherway works. I only used the YAML only because Francesco had it given already. Would it be better to ignore the file he made and go for this "manual" listing of files? I originally thought the YAML was easier because it was already available.

Reply:
The problem is that this means that you always have to update the yaml file if you add extra files to the directory. If we always need all files, then getting them out of the directory with the above method will work without changing the yaml file and will also save us the extra dependency of pyyaml. Less dependencies is usually better, although here it's probably not a big deal.